.github/workflows/checkstyle.yml

```diff
@@ -0,0 +1,27 @@
+name: checkstyle
```

Comment:
Why do you have separate workflows for the build related tasks? I would create one central `build` workflow, having different `jobs` like `checkstyle`, `test-postrgre`, `test-mysql`.

This also has the advantage that you can have a central `compile` job, and then share the build output with the different other jobs that rely on the compilation (e.g. tests / checkstyle). This would decrease build time dramatically. See https://help.github.com/en/actions/automating-your-workflow-with-github-actions/persisting-workflow-data-using-artifacts#passing-data-between-jobs-in-a-workflow